tests/platform_tests/api/test_psu.py

```diff
@@ -0,0 +1,108 @@
+from common.helpers.platform_api import chassis, psu
+
+from platform_api_test_base import PlatformApiTestBase
+
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.sanity_check(skip_sanity=True),
+    pytest.mark.disable_loganalyzer  # disable automatic loganalyzer
+]
+
+STATUS_LED_COLOR_GREEN = "green"
+STATUS_LED_COLOR_AMBER = "amber"
+STATUS_LED_COLOR_RED = "red"
+STATUS_LED_COLOR_OFF = "off"
+
+class TestPsuApi(PlatformApiTestBase):
+    ''' Platform API test cases for the PSU class'''
+
+    num_psus = None
+
+    @pytest.fixture(scope="function", autouse=True)
+    def setup(self, platform_api_conn):
+        if self.num_psus is None:
+            try:
+                self.num_psus = int(chassis.get_num_psus(platform_api_conn))
+            except:
+                pytest.fail("num_psus is not an integer")
+
+
+    def test_fans(self, duthost, localhost, platform_api_conn):
+        ''' PSU fan test '''
+        for psu_id in range(self.num_psus):
+            try:
+                num_fans = int(psu.get_num_fans(platform_api_conn, psu_id))
+            except:
+                pytest.fail("num_fans is not an integer!")
+
+            fan_list = psu.get_all_fans(platform_api_conn, psu_id)
+            if self.expect(fan_list is not None, "Failed to retrieve fans of PSU {}".format(psu_id)):
+                self.expect(isinstance(fan_list, list) and len(fan_list) == num_fans, "Fans of PSU {} appear to be incorrect".format(psu_id))
+
+            for i in range(num_fans):
+                fan = psu.get_fan(platform_api_conn, psu_id, i)
+                self.expect(fan and fan == fan_list[i], "Fan {} of PSU {} is incorrect".format(i, psu_id))
+        self.assert_expectations()
+
+
+    def test_power(self, duthost, localhost, platform_api_conn):
+        ''' PSU power test '''
+        for psu_id in range(self.num_psus):
+            voltage = psu.get_voltage(platform_api_conn, psu_id)
+            current = psu.get_current(platform_api_conn, psu_id)
+            power = psu.get_power(platform_api_conn, psu_id)
+            self.expect(abs(power - (voltage*current)) < power*0.1, "PSU {} reading does not make sense \
+                (power:{}, voltage:{}, current:{})".format(psu_id, power, voltage, current))
+
+            powergood_status = psu.get_powergood_status(platform_api_conn, psu_id)
+            if self.expect(powergood_status is not None, "Failed to retrieve operational status of PSU {}".format(psu_id)):
+                self.expect(powergood_status is True, "PSU {} is not operational".format(psu_id))
+
+            high_threshold = psu.get_voltage_high_threshold(platform_api_conn, psu_id)
+            low_threshold = psu.get_voltage_low_threshold(platform_api_conn, psu_id)
+            if self.expect(high_threshold is not None and low_threshold is not None, "Failed to retrieve the voltage threshold values of PSU {}".format(psu_id)):
```

Comment:
Suggest to split this into separate assertions for high_threshold and low_threshold so that we can get more clear error messages of which one (or both) failed. Maybe something like:

```
high_threshold = psu.get_voltage_high_threshold(platform_api_conn, psu_id)
if self.expect(high_threshold is not None, "Failed to retrieve the high voltage threshold value of PSU {}".format(psu_id)):
    self.expect(<check that high_threshold value is of proper expected type (float?) here> ...)

low_threshold = psu.get_voltage_low_threshold(platform_api_conn, psu_id)
if self.expect(low_threshold is not None, "Failed to retrieve the low voltage threshold value of PSU {}".format(psu_id)):
    self.expect(<check that low_threshold value is of proper expected type (float?) here> ...)

if high_threshold and low_threshold:
    self.expect(voltage < high_threshold and voltage > low_threshold, "Voltage {} of PSU {} is not in between {} and {}".format(voltage, psu_id, low_threshold, high_threshold))
```